.circleci/config.yml

```
@@ -1,153 +1,52 @@
-
-  checkfmt:
-    <<: *default_sbt_job
-    steps:
-      - attach_workspace:
-          at: ~/
-      - *restore_sbt_cache
-      - run:
-          name: Check formatting
-          command: sbt ";scalafmt::test;test:scalafmt::test;sbt:scalafmt::test"
-      - *clean_sbt_cache
-      - *save_sbt_cache
-
-  populate_cache:
-    <<: *default_sbt_job
-    steps:
-      - attach_workspace:
-          at: ~/
-      - *restore_sbt_cache
-      - run:
-          name: Safely populate caches
-          command: sbt ";set scalafmtUseIvy in ThisBuild := false;update"
-      - *clean_sbt_cache
-      - *save_sbt_cache
-
-  compile_and_test:
-    <<: *default_sbt_job
-    steps:
-    - attach_workspace:
-        at: ~/
-    - *restore_sbt_cache
-    - run:
-        name: Compile / Test and send coverage results
-        command: sbt ";compile;test:compile;coverage;test;coverageReport;coverageAggregate;codacyCoverage"
-    - *clean_sbt_cache
-    - *save_sbt_cache
-
-  tag_version:
-    <<: *default_sbt_job
-    steps:
-    - attach_workspace:
-        at: ~/
-    - add_ssh_keys:
-        fingerprints:
-          - "df:83:d7:c7:d5:79:06:c2:3b:d1:fd:e2:a3:d1:12:c5"
-    - run:
-        name: Tag
-        command: git tag $(cat .version) && git push --tags
-
-  publish:
-    <<: *default_sbt_job
-    steps:
-      - attach_workspace:
-          at: ~/
-      - *restore_sbt_cache
-      - run:
-          name: Publish to sonatype
-          command: sbt ";clean;retrieveGPGKeys;+publishSigned;sonatypeReleaseAll"
+orbs:
+  codacy: codacy/base@2.6.4
 
 workflows:
   version: 2
-  compile_deploy:
+  build:
     jobs:
-    - checkout_and_version
-    - populate_cache:
-        requires:
-        - checkout_and_version
-    - checkfmt:
-        requires:
-        - populate_cache
-    - compile_and_test:
-        requires:
-        - populate_cache
-    - publish:
-        filters:
-          branches:
-            only:
-            - master
-        context: CodacyAWS
-        requires:
-        - compile_and_test
-        - checkfmt
-    - tag_version:
-        filters:
-          branches:
-            only:
-            - master
-        context: CodacyAWS
-        requires:
-        - publish
+      - codacy/checkout_and_version:
+          write_sbt_version: true
+      - codacy/sbt:
+          name: populate_cache
+          persist_to_workspace: true
+          cmd: sbt "set scalafmtUseIvy in ThisBuild := false;+update"
+          requires:
+            - codacy/checkout_and_version
+      - codacy/sbt:
+          name: check_format
+          cmd: sbt "scalafmt::test;test:scalafmt::test;sbt:scalafmt::test"
+          requires:
+            - populate_cache
+      - codacy/sbt:
+          name: compile
+          persist_to_workspace: true
+          cmd: sbt "+compile;+test:compile"
+          requires:
+            - populate_cache
+      - codacy/sbt:
+          name: test
+          cmd: |
+            sbt "coverage;test;coverageReport"
```

You can even split by spaces:
```suggestion
            sbt coverage test coverageReport
```
Not sure but I think you need to run `coverageAggregate` then (in a separate command):
```
sbt coverageAggregate
```